recipes-openxt/openxt/openxt-measuredlaunch/ml-functions

```diff
@@ -384,6 +445,37 @@ generate_policy() {
     [ -f $policy ] && rm -f $policy
 }
 
+# Function to read policy from TPM NVRAM and store into a file
+# parameter 0: path where policy file should be written
+# parameter 1: hash algorithm to use, either sha1 or sha256
+# return 0 on success
+# return 1 if fails to read
+extract_policy() {
+    local policy=$1
+    local hash_alg="sha1" #default to sha1 for legacy support
+
+    is_tpm_2_0
+    tpm2=$?
+
+    if [ "${tpm2}" -eq 0 ]; then
+        hash_alg=$(get_preferred_alg)
+        [ $? -ne 0 ] && echo "No valid algs available" && return 1
+        #Two possible nv index locations on tpm2.0 that tboot checks
+        local tboot_idx='0x01200001'
```

Comment:
I only need one and this relies on the fact that you have already stored it in both locations during generate-policy. I am not worried which index was used by tboot because we put the same policy in both.